thorlcr/slave/slave.cpp

```diff
@@ -287,6 +289,18 @@ class CGenericSlaveGraphElement : public CSlaveGraphElement
         }
         haveCreateCtx = true;
     }
+    virtual CActivityBase *queryActivity()
+    {
+        if (isEof)
```

Comment:
I don't think this causes problems, but this should really be using a different variable for this (it could set isEof as well).  Otherwise if queryActivity() is called on an activity that has read all records it could return the null activity instead.

In fact could this be a problem?  If an activity in a slave graph has set isEof, when it is reset, could that cause the wrong activity to be returned?

@jakesmith